web-common/src/features/dashboards/filters/Filters.svelte

```diff
@@ -70,54 +45,15 @@ The main feature-set component for dashboard filters
   let topListQuery;
   let searchText = "";
   let searchedValues = [];
-  let activeDimensionName;
-
-  $: metricTimeSeries = useModelHasTimeSeries(
-    $runtime.instanceId,
-    metricViewName
-  );
-  $: hasTimeSeries = $metricTimeSeries.data;
-
-  $: addNull = "null".includes(searchText);
-
-  $: if (activeDimensionName) {
-    if (searchText == "") {
-      searchedValues = [];
-    } else {
-      let topListParams = {
-        dimensionName: activeDimensionName,
-        limit: "15",
-        offset: "0",
-        sort: [],
-        filter: {
-          include: [
-            {
-              name: activeDimensionName,
-              in: addNull ? [null] : [],
-              like: [`%${searchText}%`],
-            },
-          ],
-          exclude: [],
-        },
-      };
-
-      if (hasTimeSeries) {
-        topListParams = {
-          ...topListParams,
-          ...{
-            timeStart: metricsExplorer?.selectedTimeRange?.start,
-            timeEnd: metricsExplorer?.selectedTimeRange?.end,
-          },
-        };
-      }
-
-      // Use topList API to fetch the dimension names
-      // We prune the measure values and use the dimension labels for the filter
-      topListQuery = createQueryServiceMetricsViewToplist(
-        $runtime.instanceId,
-        metricViewName,
-        topListParams
-      );
+  let activeDimensionName: string;
+
+  $: {
+    if (activeDimensionName) {
```

Comment:
If the query is conditional on `activeDimensionName`, should it go in the query's `enabled` option?

Reply:
Yea totally, good call. Let me take a stab at adding that and see how it looks

Reply:
Done